app/controllers/favorites_controller.rb

```diff
@@ -2,6 +2,7 @@ class FavoritesController < ApplicationController
   def index
     @favorited_pets = Pet.find(@favorites.ids)
     @applied_pets = Pet.select(:name, :id).joins(:pet_applications).distinct
+    @approved_pets = @applied_pets.where('pet_applications.approved=true')
```

Comment:
Can we refactor active record out of the controller to the model level?

Reply:
(like we talked about together:)
I agree that we should, but not sure where else to put it: maybe a module? Maybe in ApplicationController?
At the very least, we could move some of the ActiveRecord querying to the model level - I'll do that.